MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Reflectometry/ReflCommand.h

```diff
@@ -33,12 +35,27 @@ Code Documentation is available at: <http://doxygen.mantidproject.org>
 */
 class ReflCommand {
 public:
+  ReflCommand() : m_child(){};
   virtual ~ReflCommand(){};
 
   virtual void execute() = 0;
   virtual std::string name() = 0;
   virtual std::string icon() = 0;
+  virtual bool hasChild() final { return !m_child.empty(); };
+  virtual void setChild(std::vector<std::unique_ptr<ReflCommand>> child) final {
+    m_child = std::move(child);
+  }
+  virtual const std::vector<std::unique_ptr<ReflCommand>> &getChild() final {
+    return m_child;
+  }
+	virtual bool isSeparator() final { return name().empty() && icon().empty(); }
+
+protected:
+  std::vector<std::unique_ptr<ReflCommand>> m_child;
 };
+
+typedef std::unique_ptr<ReflCommand> ReflCommand_uptr;
+typedef std::shared_ptr<ReflCommand> ReflCommand_sptr;
 }
```

Comment:
In future, consider having abstract classes that are completely abstract. That means they are purely a **contract** and contain no information about how that contract is to be provided. Then have base classes or nested functionality in sub-classes that provide the **common-code shared aspects** needed by the concrete types. Forcing this separation really does help.